src/go/rpk/pkg/cli/registry/subject.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+)
+
+func subjectCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "subject",
+		Args:  cobra.ExactArgs(0),
+		Short: "List or delete schema registry subjects",
+	}
+	cmd.AddCommand(
+		subjectListCommand(fs, p),
+		subjectDeleteCommand(fs, p),
+	)
+	p.InstallFormatFlag(cmd)
+	return cmd
+}
+
+func subjectListCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var deleted bool
+	cmd := &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls"},
+		Short:   "Display all subjects",
+		Args:    cobra.ExactArgs(0),
+		Run: func(cmd *cobra.Command, _ []string) {
+			f := p.Formatter
+			if h, ok := f.Help([]string{}); ok {
+				out.Exit(h)
+			}
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			cl, err := schemaregistry.NewClient(fs, p)
+			out.MaybeDie(err, "unable to initialize schema registry client: %v", err)
+
+			subjects, err := cl.Subjects(cmd.Context(), sr.HideShowDeleted(deleted))
+			out.MaybeDieErr(err)
+
+			if isText, _, s, err := f.Format(subjects); !isText {
+				out.MaybeDie(err, "unable to print in the required format %q: %v", f.Kind, err)
+				out.Exit(s)
+			}
+			sort.Strings(subjects)
+			for _, s := range subjects {
+				fmt.Println(s)
+			}
+		},
+	}
+	cmd.Flags().BoolVar(&deleted, "deleted", false, "If true, list deleted subjects as well")
+	return cmd
+}
+
+type deleteResponse struct {
+	Subject  string `json:"subject" yaml:"subject"`
+	Versions []int  `json:"versions_deleted,omitempty" yaml:"versions_deleted,omitempty"`
+	Err      string `json:"error,omitempty" yaml:"error,omitempty"`
+}
+
+func subjectDeleteCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var isPermanent bool
+	cmd := &cobra.Command{
+		Use:   "delete [SUBJECT...]",
+		Short: "Soft or hard deletion of subjects",
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, subjects []string) {
+			f := p.Formatter
+			if h, ok := f.Help([]deleteResponse{}); ok {
+				out.Exit(h)
+			}
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			cl, err := schemaregistry.NewClient(fs, p)
+			out.MaybeDie(err, "unable to initialize schema registry client: %v", err)
+
+			var (
+				wg      sync.WaitGroup
+				mu      sync.Mutex
+				results []deleteResponse
+			)
+
+			for i := range subjects {
+				subject := subjects[i]
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					var versions []int
+					var err error
+					if isPermanent {
+						versions, err = cl.DeleteSubject(cmd.Context(), subject, sr.SoftDelete)
+						if err == nil || schemaregistry.IsSubjectNotFoundError(err) {
+							versions, err = cl.DeleteSubject(cmd.Context(), subject, sr.HardDelete)
+							if err != nil {
+								err = fmt.Errorf("unable to perform hard-deletion: %w", err)
+							}
+						} else {
+							err = fmt.Errorf("unable to perform soft-deletion: %w", err)
```

Comment:
`unable to perform initial soft-deletion that is required for hard-deletion: %v`
^^ I think we should expand the wording a little bit because if I were to see "unable to perform soft-deletion" when I'm trying to hard delete, I'd wonder what's up